boa/src/syntax/lexer/mod.rs

```diff
@@ -423,7 +423,7 @@ impl<'a> Lexer<'a> {
                         }
                     };
 
-                    self.push_token(TokenData::NumericLiteral(num));
+                    self.push_token(TokenKind::NumericLiteral(num as f64));
```

Comment:
Why is this casting to `f64` needed?

Reply:
hmm good question, now i can't remember, i think an error happened, if it works as num we should revert it